src/material-examples/table-http/table-http-example.ts

```diff
@@ -18,20 +17,48 @@ import 'rxjs/add/operator/switchMap';
   styleUrls: ['table-http-example.css'],
   templateUrl: 'table-http-example.html',
 })
-export class TableHttpExample implements OnInit {
-  displayedColumns = ['created_at', 'state', 'number', 'title'];
-  exampleDatabase: ExampleHttpDataSource | null;
-  dataSource: ExampleDataSource | null;
+export class TableHttpExample {
+  displayedColumns = ['created', 'state', 'number', 'title'];
+  exampleDatabase: ExampleHttpDao | null;
+  dataSource = new MatTableDataSource();
+
+  resultsLength = 0;
+  isLoadingResults = false;
+  isRateLimitReached = false;
 
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
 
-  constructor(private _httpClient: HttpClient) {}
+  constructor(private http: HttpClient) {}
 
-  ngOnInit() {
-    this.exampleDatabase = new ExampleHttpDataSource(this._httpClient);
-    this.dataSource = new ExampleDataSource(
-      this.exampleDatabase!, this.paginator, this.sort);
+  ngAfterViewInit() {
+    this.exampleDatabase = new ExampleHttpDao(this.http);
+
+    // If the user changes the sort order, reset back to the first page.
+    this.sort.sortChange.subscribe(() => this.paginator.pageIndex = 0);
+
+    Observable.merge(...[this.sort.sortChange, this.paginator.page])
+        .startWith(null)
+        .switchMap(() => {
+          this.isLoadingResults = true;
+          return this.exampleDatabase!.getRepoIssues(
+              this.sort.active, this.sort.direction, this.paginator.pageIndex);
+        })
+        .map(data => {
+          // Flip flag to show that loading has finished.
+          this.isLoadingResults = false;
+          this.isRateLimitReached = false;
+          this.resultsLength = data.total_count
+
+          return data.items;
+        })
```

Comment:
I like `do` better here but concerned that adding more operators will distract people. Hard balance to find since people will probably copy/paste this =P

Reply:
Yeah, agreed on it being distracting. `map` works just as well since you still do have to map it to `items`.